src/actions/profile-view.js

```diff
@@ -44,21 +55,131 @@ import type {
 import type { Transform } from '../types/transforms';
 
 /**
- * The actions that pertain to changing the view on the profile, including searching
- * and filtering. Currently the call tree's actions are in this file, but should be
- * split apart. These actions should most likely affect every panel.
+ * This file contains actions that pertain to changing the view on the profile, including
+ * searching and filtering. Currently the call tree's actions are in this file, but
+ * should be split apart. These actions should most likely affect every panel.
+ */
+
+/**
+ * Select a call node for a given thread. An optional call node path can be provided
+ * to expand child nodes beyond the selected call node path.
  */
 export function changeSelectedCallNode(
   threadIndex: ThreadIndex,
-  selectedCallNodePath: CallNodePath
+  selectedCallNodePath: CallNodePath,
+  optionalExpandedToCallNodePath?: CallNodePath
 ): Action {
+  if (optionalExpandedToCallNodePath) {
+    for (let i = 0; i < selectedCallNodePath.length; i++) {
+      if (selectedCallNodePath[i] !== optionalExpandedToCallNodePath[i]) {
+        // This assertions ensures that the selectedCallNode will be correctly expanded.
+        throw new Error(
+          oneLine`
+            The optional expanded call node path provided to the changeSelectedCallNode
+            must contain the selected call node path.
+          `
+        );
+      }
+    }
+  }
   return {
     type: 'CHANGE_SELECTED_CALL_NODE',
     selectedCallNodePath,
+    optionalExpandedToCallNodePath,
     threadIndex,
   };
 }
 
+/**
+ * Given a sample index, select the leaf most call node for a thread.
```

Comment:
"leaf most call node for a thread" sounds odd. How about:
`Given a threadIndex and a sampleIndex, select the call node at the top ("leaf") of that sample's stack.`